app/code/core/Mage/ConfigurableSwatches/Block/Catalog/Media/Js/List.php

```diff
@@ -64,4 +64,22 @@ protected function _getImageSizes()
     {
         return ['small_image'];
     }
+
+    /**
+     * Prevent actual block render if we are disabled, and i.e. via the module
+     * config as opposed to the advanced module settings page
+     *
+     * @return string
+     */
+    protected function _toHtml()
+    {
+        if (!Mage::helper('configurableswatches')->isEnabled()) { // functionality disabled
+            return ''; // do not render block
+        }
```

Comment:
Yeah perhaps it's a performance shortcut so I left it, but indeed it could functionally be removed

Reply:
Since the parent is also the parent of other child classes and perhaps customizations, I don't want to touch it. So we could take out this check from `List.php` because the config check will not be expensive.